### PR TITLE
OHRM5X-1814: Develop Buzz - Update Post API

### PR DESCRIPTION
**Pull request by ManulMax**

### CODE REVIEW

File: src/plugins/orangehrmBuzzPlugin/test/Api/BuzzPostAPITest.php [hunk 1]
```diff
@@ -75,4 +75,36 @@ public function testGetValidationRuleForDelete(): void
...
+    {
+        $this->populateFixtures('BuzzPostAPITest.yaml');
+        $this->createKernelWithMockServices([Services::AUTH_USER => $this->getMockAuthUser($testCaseParams)]);
+        $this->registerServices($testCaseParams);
+        $this->registerMockDateTimeHelper($testCaseParams);
+        $api = $this->getApiEndpointMock(BuzzPostAPI::class, $testCaseParams);
+        $this->assertValidTestCase($api, 'update', $testCaseParams);
+    }
+
+    public function dataProviderForTestUpdate()
```
Comment (RajithaKumara) [inline src/plugins/orangehrmBuzzPlugin/test/Api/BuzzPostAPITest.php:106]:
```suggestion
    public function dataProviderForTestUpdate(): array
```

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

File: src/plugins/orangehrmBuzzPlugin/Dto/BuzzVideoURL/BuzzVideoURL.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Buzz\Dto\BuzzVideoURL;
+
+interface BuzzVideoURL
```
[the remark this answers is not in the document]
Comment (ManulMax) [reply src/plugins/orangehrmBuzzPlugin/Dto/BuzzVideoURL/BuzzVideoURL.php:22]: No, Not sure how this came here

[the remark this answers is not in the document]
Comment (ManulMax) [reply src/plugins/orangehrmBuzzPlugin/Dto/BuzzVideoURL/BuzzVideoURL.php:22]: No, Not sure how this came here

File: src/plugins/orangehrmBuzzPlugin/Api/BuzzURLValidationAPI.php [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+use OrangeHRM\Buzz\Api\ValidationRules\BuzzVideoLinkValidationRule;
+use OrangeHRM\Core\Api\V2\CollectionEndpoint;
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+
+class BuzzURLValidationAPI extends Endpoint implements CollectionEndpoint
```
[the remark this answers is not in the document]
Comment (ManulMax) [reply src/plugins/orangehrmBuzzPlugin/Api/BuzzURLValidationAPI.php:30]: No this also come accidentally

File: src/plugins/orangehrmBuzzPlugin/Api/BuzzPostAPI.php [hunk 4]
```diff
@@ -334,6 +347,7 @@ private function getTextValidationRuleForTextPost(): ?ParamRule
         return new ParamRule(
             self::PARAMETER_POST_TEXT,
             new Rule(Rules::STRING_TYPE),
+            new Rule(Rules::REQUIRED),
```
Comment (ManulMax) [inline src/plugins/orangehrmBuzzPlugin/Api/BuzzPostAPI.php:350]: TODO

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

**Pull request merged**